File: Exercism Codes/Python/isbn_verifier.py
def is_valid(isbn):
    isbn_10 = isbn.replace("-", "")
    n = len(isbn_10)
    if n != 10:
        return False

    total = 0
    for idx, digit in enumerate(isbn_10):
        m = n - idx
        if idx == n - 1 and digit == "X":
            total += 10 * m
            continue

        if digit.isdigit():
            total += int(digit) * m
        else:
            return False

    return total % 11 == 0

File: Exercism Codes/JavaScript/isbn-verifier.js
export const isValid = (isbn = "") => {
  const isbn10 = isbn.replace(/-/g, '');
  if (isbn10.length != 10) {
    return false;
  }
  let total = 0;
  for (let i = 0; i < isbn10.length; i++) {
    const n = isbn10.length - i;
    if (i === isbn10.length - 1 && isbn10[i] === "X") {
      total += 10 * n;
      continue;
    }
    if (/\d/.test(isbn10[i])) {
      total += parseInt(isbn10[i]) * n;
    }
    else {
      return false;
    }
  }
  return total % 11 === 0;
};